### PR TITLE
storage: add tests for pipelined pessimistic lock

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?
Problem Summary:
Tests for https://github.com/tikv/tikv/pull/6984 are not enough.

### What is changed and how it works?
What's Changed:
add tests for pipelined pessimistic lock

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
* No release note

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -482,6 +484,16 @@ impl<S: Snapshot> MvccTxn<S> {
             .into());
         }
         if let Some((commit_ts, _)) = self.reader.seek_write(key, TimeStamp::max())? {
+            // The invariants of pessimistic locks are:
+            //   1. lock's for_update_ts > key's latest commit_ts
+            //   2. lock's for_update_ts >= txn's start_ts
+            //   3. If the data is changed, key's new commit_ts > lock's for_update_ts
```
Comment (nrc) [inline src/storage/mvcc/txn.rs:490]: Seems 1 and 3 contradict, is 1 true except when handling new data?

Comment (youjiali1995) [reply src/storage/mvcc/txn.rs]: They are not contradict, but the comments are not clear. If succeeds in acquiring pessimistic locks, 1 is true. If the data is changed after acquiring pessimistic locks(dut to pipelining procedure), 3 is true.

**Comment by youjiali1995** [comment]
@HunDunDM PTAL

**Pull request merged**